src/os/bluestore/BlueStore.cc

```diff
@@ -5943,8 +5943,12 @@ int BlueStore::_do_alloc_write(
       compressed_bl.claim_append(t);
       uint64_t rawlen = compressed_bl.length();
       uint64_t newlen = ROUND_UP_TO(rawlen, min_alloc_size);
-      if (newlen < final_length) {
-	// pad out to min_alloc_size
+      uint64_t dstlen = final_length *
+        g_conf->bluestore_compression_required_ratio;
+      dstlen = ROUND_UP_TO(dstlen, min_alloc_size);
+      if (newlen < dstlen) {
```

Comment:
i would do <=, as .125 is 1/8 and it is likely we'll actually hit that pretty frequently.  maybe also keep newlen < final_length in there too, as the configured min after rounding up might == final_length?